client/src/main/java/tds/config/TimeLimitConfiguration.java

```diff
@@ -0,0 +1,183 @@
+ *
+ *              # clientname, _efk_testid, environment
+ *              'SBAC_PT', NULL, NULL
+ *              'SBAC', NULL, NULL
+ *
+ *              SELECT clientname, _efk_testid, environment FROM session.timelimits;
+ *
+ *              # clientname, _efk_testid, environment
+ *              'SBAC', NULL, 'Development'
+ *              'SBAC_PT', NULL, 'Development'
+ *          }
+ *     <pre>
+ *
+ *     For reasons that are not entirely clear, The legacy code specifically references the {@code session._externs}
+ *     table for the client name and environment fields, so that behavior has been carried forward.
+ * </p>
+ */
+public class TimeLimitConfiguration {
+    private String clientName;
+    private String environment;
+    private String assessmentId;
+    private int examRestartWindowMinutes;
+    private int examDelayDays;
+    private Integer interfaceTimeoutMinutes;
+    private int requestInterfaceTimeoutMinutes;
+    private Integer taCheckinTimeMinutes;
+
+    public static class Builder {
+        private String clientName;
+        private String environment;
+        private String assessmentId;
+        private int examRestartWindowMinutes;
+        private int examDelayDays;
+        private Integer interfaceTimeoutMinutes;
+        private int requestInterfaceTimeoutMinutes;
+        private Integer taCheckinTimeMinutes;
+
+        public Builder withClientName(String clientName) {
+            this.clientName = clientName;
+            return this;
+        }
+
+        public Builder withEnvironment(String environment) {
+            this.environment = environment;
+            return this;
+        }
+
+        public Builder withAssessmentId(String assessmentId) {
+            this.assessmentId = assessmentId;
+            return this;
+        }
+
+        public Builder withExamRestartWindowMinutes(int examRestartWindowMinutes) {
+            this.examRestartWindowMinutes = examRestartWindowMinutes;
+            return this;
+        }
+
+        public Builder withExamDelayDays(int newExamDelayDays) {
+            this.examDelayDays = newExamDelayDays;
+            return this;
+        }
+
+        public Builder withInterfaceTimeoutMinutes(Integer interfaceTimeoutMinutes) {
+            this.interfaceTimeoutMinutes = interfaceTimeoutMinutes;
+            return this;
+        }
+
+        public Builder withRequestInterfaceTimeoutMinutes(int requestInterfaceTimeoutMinutes) {
+            this.requestInterfaceTimeoutMinutes = requestInterfaceTimeoutMinutes;
+            return this;
+        }
+
+        public Builder withTaCheckinTimeMinutes(Integer taCheckinTimeMinutes) {
+            this.taCheckinTimeMinutes = taCheckinTimeMinutes;
+            return this;
+        }
+
+        public TimeLimitConfiguration build() {
+            return new TimeLimitConfiguration(this);
+        }
+    }
+
+    public TimeLimitConfiguration() {}
```

Comment:
When is this needed?  If it isn't leveraged via reflection or by a calling class, which I would find odd, I suggest removing it.